### ISSUE TITLE
ValuePolicy type stability

### ISSUE THREAD

**Issue by zsunberg** [JuliaPOMDP/POMDPToolbox.jl#61]
Right now, the `mdp` field of `ValuePolicy` has an abstract type. This will probably result in inefficient dynamic function calls. It should be made  concrete through parameterization.

**Comment by MaximeBouton** [comment]
I can submit a PR with something like this:
```julia 
mutable struct ValuePolicy{A, P <: Union{MDP, POMDP}} <: Policy
    mdp::P
    value_table::Matrix{Float64}
    act::Vector{A}
end
function ValuePolicy{P <: Union{MDP, POMDP}}(mdp::P)
    A = action_type(mdp)
    return ValuePolicy{A, P}(mdp, zeros(n_states(mdp), n_actions(mdp)), ordered_actions(mdp))
end
```

I am not sure how to check for type stability, I tried to use `@code_warntype` as follows to check for type instability:
```julia
using POMDPs, POMDPToolbox, POMDPModels
gw = GridWorld()
ValuePolicy(gw) # compile first
@code_warntype ValuePolicy(gw)
```
but the output is somewhat hard to interpret.

**Comment by MaximeBouton** [comment]
Sounds good, I will try Traceur and submit a PR.

**Comment by MaximeBouton** [comment]
fixed by #65
